draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -5307,14 +5307,22 @@ Expect: 100-continue
    bookmarks/favorites), the user agent &MUST; either exclude the Referer header field
    or send it with a value of "about:blank".
 </t>
+<t>
+   The Referer header field value need not convey the full URI of the referring
+   page; it &MAY; be truncated to the referring origin.
+</t>
 <t>
    The Referer header field has the potential to reveal information about the request
    context or browsing history of the user, which is a privacy concern if the
    referring resource's identifier reveals personal information (such as an
    account name) or a resource that is supposed to be confidential (such as
    behind a firewall or internal to a secured service). Most general-purpose
    user agents do not send the Referer header field when the referring
-   resource is a local "file" or "data" URI. A user agent &MUST-NOT; send a
+   resource is a local "file" or "data" URI. A user agent &SHOULD-NOT; send a
+   <x:ref>Referer</x:ref> header field if the referring page was received with
+   a secure protocol, and the request target has an origin differing from that
+   of the referring page, unless the referring page explicitly allows it.
```

Comment:
The "referring page" is a bit of a weird concept here.  I know that the spec is bad about letting browser UI leak into the language in places, but it would be good if it didn't have to here.

In essence, it's the referring resource that makes the call here, but it's a little indirect even then.  Can this just be:

```suggestion
   of the referring resource, unless the referring resource explicitly allows it.
```

Using resource throughout is probably OK, even allowing for the fact that referrer policy isn't necessarily expressed by the referring resource.